language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/CompletionUtil.java

```diff
@@ -138,4 +145,27 @@ public static void fillTokenInfoAtCursor(BallerinaCompletionContext context) {
 
         context.setNodeAtCursor(nonTerminalNode);
     }
+
+    /**
+     * Check whether the cursor is within a comment.
+     */
+    private static boolean isWithinComment(BallerinaCompletionContext ctx) {
+        Iterator<Minutiae> minutiaeIterator = Collections.emptyIterator();
+        if (ctx.getCursorPositionInTree() <= ctx.getTokenAtCursor().textRange().startOffset()) {
+            minutiaeIterator = ctx.getTokenAtCursor().leadingMinutiae().iterator();
+        } else if (ctx.getTokenAtCursor().textRange().endOffset() <= ctx.getCursorPositionInTree()) {
+            minutiaeIterator = ctx.getTokenAtCursor().trailingMinutiae().iterator();
+        }
+
+        while (minutiaeIterator.hasNext()) {
+            Minutiae minutiae = minutiaeIterator.next();
+            if (minutiae.kind() == SyntaxKind.COMMENT_MINUTIAE
+                    && minutiae.lineRange().startLine().line() == ctx.getCursorPosition().getLine()
+                    && minutiae.textRange().startOffset() < ctx.getCursorPositionInTree()
+                    && ctx.getCursorPositionInTree() <= minutiae.textRange().endOffset()) {
```

Comment:
Just out of curiosity, can't we use the offset check here?

Reply:
Makes sense. Removed the line check 👍

Reply:
@TharushiJay @nadeeshaan I'm ok with the change as long as a comment minutiae doesn't expand multiple lines. Let's wait for the tests to pass/fail since we have a test covering this scenario.